tools/tpm2_createpolicy.c

```diff
@@ -37,266 +37,101 @@
-
-    if (ctx->pcr_flags.policy_type_pcr_flag) {
-        if (!ctx->policyfile || !ctx->pcr_flags.pcr_index_flag) {
-            showArgMismatch(argv[0]);
-            return false;
-        } else {
-            // pcr_index is unsigned... never can be less than 0
-            TPM_RC rval = get_max_supported_pcrs(ctx->sapi_context,
-                &ctx->max_supported_pcrs);
-            if(rval != TPM_RC_SUCCESS) {
-                LOG_ERR("Failure to read the capability data from TPM.\n");
-                return false;
-            }
-            if (!ctx->max_supported_pcrs) {
-                LOG_ERR("Failed to retrieve number of supported PCRs on the TPM\n");
-                return false;
-            }
-
-            if (ctx->pcr_index > (ctx->max_supported_pcrs - 1)) {
-                LOG_ERR("Invalid pcr_index %u. Choose between 0..%d\n",
-                        ctx->pcr_index, ctx->max_supported_pcrs);
-                return false;
-            }
-
-            LOG_INFO("Policy File = %s\n", ctx->policyfile);
-            LOG_INFO("PCR Index= %d\n", ctx->pcr_index);
-            rval = build_policy(ctx, build_pcr_policy);
-            if (rval != TPM_RC_SUCCESS) {
-                LOG_ERR("Failed build_policy\n");
-                return false;
-            }
-        }
-    }
-    return true;
-}
-
-static bool init(int argc, char *argv[], createpolicypcr_ctx *ctx) {
-
+static bool init(int argc, char *argv[], CREATE_POLICY_CTX *pctx) {
     struct option sOpts[] = { 
-        { "policy-file",    required_argument,  NULL,   'f' }, 
-        { "policy-pcr",     no_argument,        NULL,  'P' }, 
-        { "pcr-index",      required_argument,  NULL,   'i' }, 
-        { "pcr-alg",        required_argument,  NULL,   'g' }, 
-        { "pcr-input-file", required_argument,  NULL,   'F' }, 
+        { "policy-file",    required_argument,  NULL,   'f' },
+        { "policy-digest-alg", required_argument, NULL, 'g'},
+        { "set-list",       required_argument,  NULL,   'L' }, 
+        { "pcr-input-file", required_argument,  NULL,   'F' },
+        { "policy-pcr",     no_argument,        NULL,   'P' }, 
+        { "auth-policy-session", no_argument, NULL,     'a'},
+        { "extend-policy-session", no_argument, NULL,   'e'},
         { NULL,             no_argument,        NULL,   '\0'}, 
     };
-
     if (argc == 1) {
         showArgMismatch(argv[0]);
         return false;
     }
-
     if (argc > (int) (2 * sizeof(sOpts) / sizeof(struct option))) {
         showArgMismatch(argv[0]);
         return false;
     }
 
     int opt;
-    bool result;
-    while ((opt = getopt_long(argc, argv, "Pf:i:g:F:", sOpts, NULL)) != -1) {
+    while ((opt = getopt_long(argc, argv, "f:g:L:F:Pae", sOpts, NULL)) != -1) {
         switch (opt) {
         case 'f':
-            ctx->policyfile = optarg;
+            pctx->common_policy_options.policy_file_flag = true;
+            snprintf(pctx->common_policy_options.policy_file, 
```

Comment:
char * here, no more of this snprintf() copying. See: https://github.com/01org/tpm2.0-tools/commit/11facc457f39659709ca29ec91327a6bb289f616

This would re-open bug: #260